app/javascript/app/components/nav/nav.js

```diff
@@ -1,12 +1,44 @@
+import { createElement, PureComponent } from 'react';
 import { connect } from 'react-redux';
-
+import PropTypes from 'prop-types';
+import { setUser, navigateToLogin } from '../../pages/login/login-actions';
+import { logout } from '../../services/login.service';
 import Component from './nav-component';
 
-const actions = null;
-// const actions = [{ name: 'Setup' }];
-
-const mapStateToProps = () => ({
-  actions
+const mapStateToProps = state => ({
+  location: state.location
 });
 
-export default connect(mapStateToProps, null)(Component);
+class NavContainer extends PureComponent {
+  handleLogout = () => {
+    const { dispatch } = this.props;
+    logout().then(ok => {
+      if (ok) {
+        dispatch(setUser({}));
+        localStorage.setItem('CWTTT', '');
+        localStorage.setItem('user', '');
+        dispatch(navigateToLogin());
+      }
+    });
+  };
+
+  render() {
+    const actions = [
```

Comment:
Having this in the render we are generating a new array each time, couldn't we create in the constructor or static property?